### PR TITLE
Replace "canned pet food" with proper pet food in a can.

### LINKED ISSUE #32911: Activating canned dogfood results in eating it
Instead of feeding the nearby dog.

0.D-6156-gcb9471dd46-dirty

### PR DESCRIPTION
**Pull request by jeremyshannon**
#### Summary
```SUMMARY: Bugfixes "Replaces the canned_dogfood and canned_catfood items with the dogfood and catfood items (coded to be usable for pets) in a can."```

#### Purpose of change

Fixes #32911
 Also fixes a problem I spotted in testing where crafting "canned dog food" gave you the unusable canned version of the food, with no can.
 As a side thing I dropped the spawn of dog food in the k9 police cruiser, because while I could be wrong, I'm pretty sure k9 officers' diets are controlled and they eat at scheduled times, at the station. They don't, like, pull over and feed the dog on the side of the road or whatever.

#### Describe the solution

Replaced all instances of canned_dogfood and canned_catfood with dogfood and catfood in cans; also added a migration entry so it should automatically replace the erroneous canned_*food items in save games. Tested crafting and spawns in pet stores, everything seems to work.

#### Describe alternatives you've considered

I could have altered the item activation code to allow canned_*food to be fed to pets like the regular stuff, but this seemed like a more proper solution to me.

### CODE REVIEW

**Comment by LaVeyanFiend** [comment]
_Glances at title of PR_
Canned pet _foot?_

**Comment by RarkGrames** [comment]
> 
> 
> _Glances at title of PR_
> Canned pet _foot?_ rofl

_"Ancient chinese shecret"_ :smile:

**Comment by jeremyshannon** [comment]
I don't know how I typoed that!

**Comment by ZhilkinSerg** [comment]
Please use [web-linter](http://dev.narc.ro/cataclysm/format.html) to lint json files.

File: data/json/itemgroups/food.json [hunk 1]
```diff
@@ -93,3 +102,4 @@
     ]
   }
 ]
+
```
Comment (ZhilkinSerg) [inline data/json/itemgroups/food.json]:
```suggestion
```

File: data/json/itemgroups/locations.json [hunk 2]
```diff
@@ -1430,3 +1436,4 @@
     "items": [ [ "duffelbag", 5 ], [ "backpack", 5 ], [ "bag_plastic", 30 ], [ "basket_laundry", 30 ] ]
   }
 ]
+
```
Comment (ZhilkinSerg) [inline data/json/itemgroups/locations.json]:
```suggestion
```

File: data/json/items/comestibles/other.json [hunk 3]
```diff
@@ -696,3 +674,4 @@
     "volume": 1
   }
 ]
+
```
Comment (ZhilkinSerg) [inline data/json/items/comestibles/other.json]:
```suggestion
```

File: data/json/items/migration.json [hunk 4]
```diff
@@ -623,3 +635,4 @@
     "replace": "reloaded_270win_jsp"
   }
 ]
+
```
Comment (ZhilkinSerg) [inline data/json/items/migration.json]:
```suggestion
```

File: data/json/mapgen/animalshelter.json [hunk 5]
```diff
@@ -316,3 +316,4 @@
     }
   }
 ]
+
```
Comment (ZhilkinSerg) [inline data/json/mapgen/animalshelter.json]:
```suggestion
```

File: data/json/mapgen/prison_1.json [hunk 6]
```diff
@@ -599,3 +599,4 @@
     ]
   }
 ]
+
```
Comment (ZhilkinSerg) [inline data/json/mapgen/prison_1.json]:
```suggestion
```

File: data/json/npcs/items_generic.json [hunk 7]
```diff
@@ -1129,3 +1129,4 @@
     ]
   }
 ]
+
```
Comment (ZhilkinSerg) [inline data/json/npcs/items_generic.json]:
```suggestion
```

File: data/json/recipes/recipe_food.json [hunk 8]
```diff
@@ -8670,3 +8672,4 @@
     ]
   }
 ]
+
```
Comment (ZhilkinSerg) [inline data/json/recipes/recipe_food.json]:
```suggestion
```

File: data/json/vehicles/emergency.json [hunk 9]
```diff
@@ -1384,3 +1382,4 @@
     ]
   }
 ]
+
```
Comment (ZhilkinSerg) [inline data/json/vehicles/emergency.json]:
```suggestion
```

File: data/json/vehicles/trucks.json [hunk 10]
```diff
@@ -891,11 +891,12 @@
...
-      { "x": -4, "y": 1, "chance": 10, "items": [ "dogfood_canned" ] },
+      { "x": -4, "y": 0, "chance": 10, "items": { "item": "dog_food", "container-item": "can_food" } },
+      { "x": -4, "y": 0, "chance": 10, "items": { "item": "dog_food", "container-item": "can_food" } },
+      { "x": -4, "y": 1, "chance": 10, "items": { "item": "cat_food", "container-item": "can_food" } },
+      { "x": -4, "y": 1, "chance": 10, "items": { "item": "cat_food", "container-item": "can_food" } },
       { "x": -4, "y": 0, "chance": 3, "items": [ "beartrap" ] }
     ]
   }
 ]
+
```
Comment (ZhilkinSerg) [inline data/json/vehicles/trucks.json]:
```suggestion
```

**Comment by ZhilkinSerg** [comment]
Did you add these empty new lines by yourself? Linter usually doesn't add them (at least there is a bug).

**Comment by jeremyshannon** [comment]
> Did you add these empty new lines by yourself? Linter usually doesn't add them (at least there is a bug).
They came straight out of the web linter for whatever reason. I'll probably use the command line linter in the future, now that I remember it exists.

**Comment by jeremyshannon** [comment]
I guess that doesn't work. Stupid git, be more smarter!

**Pull request merged**